aiidalab_widgets_base/elns.py

```diff
@@ -0,0 +1,285 @@
+import json
+from pathlib import Path
+
+import ipywidgets as ipw
+import requests_cache
+import traitlets
+from aiida.orm import Node, QueryBuilder
+from aiidalab_eln import get_eln_connector
+from IPython.display import clear_output, display
+
+ELN_CONFIG = Path.home() / ".aiidalab" / "aiidalab-eln-config.json"
+ELN_CONFIG.parent.mkdir(
+    parents=True, exist_ok=True
+)  # making sure that the folder exists.
+
+
+def connect_to_eln(eln_instance=None, **kwargs):
+    try:
+        with open(ELN_CONFIG, "r") as file:
+            config = json.load(file)
+    except (FileNotFoundError, json.JSONDecodeError, KeyError):
+        return None
```

Comment:
The warning message `""Warning! The access to ELN {kwargs['eln_instance']} is not configured. Please follow <a href="{url}" target="_blank">the link</a> to configure it.` users get here (in case reading the config fails and you return `None`) might not be so clear.

Reply:
We had a separate discussion with @kjappelbaum regarding this comment. It turns out that if the file is erroneously formatted - it will be erased. So the warning message still holds since the user will have to configure the ELN access again. One might argue that this is not the best behaviour, but it looks reasonable to us.